### ISSUE TITLE
Clarification on documentation: does `stumpy.match` rely on `MPDist`?

### ISSUE THREAD

**Issue by patrickfleith** [TDAmeritrade/stumpy#798]
I would like to understand better how the methods `stumpy.match` and `stumpy.mass` compute the distance under the hood to find the best matches. I assume it relies on the measure `MPDist` but it is not explicitly mention in the documentation which distance measure it uses. Could you clarify?

**Comment by seanlaw** [comment]
Hi @patrickfleith! Welcome to the STUMPY community and thank you for your question.

> I would like to understand better how the methods stumpy.match and stumpy.mass compute the distance under the hood to find the best matches. I assume it relies on the measure MPDist but it is not explicitly mention in the documentation which distance measure it uses. Could you clarify?

No, `stumpy.match` uses `stumpy.mass` to compute the distance under the hood. For all intents and purposes, `stumpy.mass` is just a fast way (leveraging FFT) to compute the sliding window dot product between the query subsequence, `Q`, and all subsequences within your time series, `T`. Once the dot products are computed, it is trivial to compute the correspodning (z-normalized) Euclidean distance(s). I recommend looking at [this paper](https://www.cs.ucr.edu/~eamonn/PID4481997_extend_Matrix%20Profile_I.pdf) to better understand the inner workings. At the end of the day, it computes the (z-normalized) Euclidean distance and that is all you should need to care about.

To be explicit, `stumpy.match` and `stumpy.mass` do not use `MPdist` and is mostly unrelated. I hope that helps.

**Comment by patrickfleith** [comment]
Great, many thanks. And from what I observe there is no other routine which search for the best matching based on MPDsit, right? One would have to come up with his.her own search function?

**Comment by seanlaw** [comment]
> Great, many thanks. And from what I observe there is no other routine which search for the best matching based on MPDsit, right? One would have to come up with his.her own search function?

That is correct. At the end of the day, you are computing a "distance" and then it is up to you to set a distance threshold that decides whether something is "significant" or not.

**Comment by seanlaw** [comment]
@patrickfleith In the future, would you mind posting usage questions to our [Github Discussions](https://github.com/TDAmeritrade/stumpy/discussions)? This way, others may chime in and provide their perspective too.
